File: dispenser_plain/dispenser.erl
-module(dispenser).
-compile(export_all).

loop(TicketId) ->
  receive
    {From, Ref, take_ticket} ->
      From ! {self(), Ref, TicketId},
      loop(TicketId + 1);
    {From, Ref, current_ticket} ->
      From ! {self(), Ref, TicketId},
      loop(TicketId);
    {reset} ->
      loop(1);
    {stop} ->
      true
  end.

%%%API
start() ->
  spawn(?MODULE, loop, [1]).

take_ticket(Pid) ->
  Ref = make_ref(),
  Pid ! {self(), Ref, take_ticket},
  receive
    {Pid, Ref, Msg} -> Msg
  end.

current_ticket(Pid) ->
  Ref = make_ref(),
  Pid ! {self(), Ref, current_ticket},
  receive
    {Pid, Ref, Msg} -> Msg
  end.

reset(Pid) ->
  Pid ! {reset}.

stop(Pid) ->
  Pid ! {stop}.

File: dispenser_plain_elixir/dispenser.ex
defmodule Dispenser do
  def start() do
    Dispenser.Server.start()
  end

  def current_ticket(pid) do
    Dispenser.Server.current_ticket(pid)
  end

  def take_ticket(pid) do
    Dispenser.Server.take_ticket(pid)
  end

  def reset(pid) do
    Dispenser.Server.reset(pid)
  end

  def stop(pid) do
    Dispenser.Server.stop(pid)
  end
end
